mapic/mistapiconnector_app.go

```diff
@@ -509,43 +606,62 @@ func (mc *mac) getStreamInfo(playbackID string) (*streamInfo, error) {
 	info := mc.streamInfo[playbackID]
 	mc.mu.RUnlock()
 
-	if info != nil {
-		return info, nil
+	if info == nil {
+		return mc.refreshStreamInfo(playbackID)
 	}
+	return info, nil
+}
+
+func (mc *mac) refreshStreamInfo(playbackID string) (*streamInfo, error) {
+	glog.Infof("Refreshing stream info for playbackID=%s", playbackID)
+	mc.mu.Lock()
+	info, infoExists := mc.streamInfo[playbackID]
+	mc.mu.Unlock()
 
-	glog.Infof("getStreamInfo: Fetching stream not found in memory. playbackID=%s", playbackID)
 	stream, err := mc.lapi.GetStreamByPlaybackID(playbackID)
 	if err != nil {
 		return nil, fmt.Errorf("error getting stream by playback ID %s: %w", playbackID, err)
 	}
 
-	pushes := make(map[string]*pushStatus)
+	newPushes := make(map[string]*pushStatus)
 	for _, ref := range stream.Multistream.Targets {
 		target, pushURL, err := mc.getPushUrl(stream, &ref)
 		if err != nil {
-			return nil, err
+			glog.Errorf("error fetching multistream target, err=%v", err)
+			continue
 		}
-		pushes[pushURL] = &pushStatus{
+		newPushes[pushURL] = &pushStatus{
 			target:  target,
 			profile: ref.Profile,
 			// Assume setup was all successful
 			pushStartEmitted: true,
+			metrics:          &data.MultistreamMetrics{},
+		}
+		if infoExists {
+			info.mu.Lock()
+			push, pushExists := info.pushStatus[pushURL]
+			info.mu.Unlock()
+			if pushExists {
+				newPushes[pushURL].metrics = push.metrics
+				newPushes[pushURL].pushStopped = push.pushStopped
+				newPushes[pushURL].pushStartEmitted = push.pushStartEmitted
```

Comment:
Shouldn't we emit new push-start events if a push was disabled and then re-enabled? Or is this logic for some other case?